File: Codes/Day2/Combination.java
class Solution {
    public static void generate(List<List<Integer>> result, List<Integer> list, int k, int n, int start) {
        if(k == 1 && start <= n && n <= 9) {
            list.add(n);
            result.add(new ArrayList<>(list));
            list.remove(Integer.valueOf(n));
            return;
        }
        for(int i = start; i <= 9; i++) {
            if(i <= n && n - i > i) {
                list.add(i);
                generate(result, list, k - 1, n - i, i + 1);
                list.remove(Integer.valueOf(i));
            }
            else break;
        }
    }
    public List<List<Integer>> combinationSum3(int k, int n) {
        List<List<Integer>> result = new ArrayList<>();
        List<Integer> list = new ArrayList<>();
        generate(result, list, k, n, 1);
        return result;
    }
}

File: Codes/Day2/Combination.cpp
class Solution {
public:
    void solve(int index, vector<int>& ds, int target, int size, vector<vector<int>>& result, vector<int> arr){
        if(ds.size() == size){
            if(target == 0) result.push_back(ds);
            return;
        }
        
        if(index == 9) return;
        if(arr[index] > target) return;

        ds.push_back(arr[index]);
        solve(index + 1, ds, target - arr[index], size, result, arr);
        ds.pop_back();
        solve(index + 1, ds, target, size, result, arr);
    }
    vector<vector<int>> combinationSum3(int k, int n) {
        vector<int> arr = {1,2,3,4,5,6,7,8,9};
        vector<int> ds;
        vector<vector<int>> result;
        solve(0, ds, n, k, result, arr);
        return result; 
    }
};
